variables.go

```diff
@@ -57,7 +57,7 @@ func (e *Executor) compiledTask(call *ast.Call, evaluateShVars bool) (*ast.Task,
 		Dir:                  templater.Replace(origTask.Dir, cache),
 		Set:                  origTask.Set,
 		Shopt:                origTask.Shopt,
-		Vars:                 nil,
```

Comment:
I've noticed this before and wondered the same thing. However. it never caused any issues until now. I have no idea if there is a reason to assign `nil` here. @andreynering are you aware of anything?

Otherwise LGTM.

Reply:
I think it was just not used at all before, because all the templating happens inside function. Looks fine to set it here.